pkg/dlna/service.go

```diff
@@ -43,50 +44,66 @@ type Service struct {
 	stopTime   *time.Time
 }
 
-func (s *Service) init() {
+func (s *Service) getInterfaces() ([]net.Interface, error) {
+	var ifs []net.Interface
+	var err error
+	ifNames := s.config.GetDLNAInterfaces()
+
+	if len(ifNames) == 0 {
+		ifs, err = net.Interfaces()
+	} else {
+		for _, n := range ifNames {
+			if_, err := net.InterfaceByName(n)
+			if err != nil {
+				return nil, fmt.Errorf("error getting interface for name %s: %s", n, err.Error())
+			}
+
+			if if_ != nil {
+				ifs = append(ifs, *if_)
+			}
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	var tmp []net.Interface
+	for _, if_ := range ifs {
+		if if_.Flags&net.FlagUp == 0 || if_.MTU <= 0 {
+			continue
+		}
+		tmp = append(tmp, if_)
+	}
+	ifs = tmp
```

Comment:
Is a `tmp` interface slice needed to clean the ifs ?
Cant we remove the above code and just adjust line 61?
```golang
if if_ != nil {
  if if_.Flags&net.FlagUp == 0 || if_.MTU <= 0 {
 	continue
  }	
ifs = append(ifs, *if_)
}

			
```

Reply:
Line 61 only applies where `ifNames` is not empty.